### PR TITLE
Refactor model generation in rlc

### PR DESCRIPTION
**Pull request by qiaozha**
fix https://github.com/Azure/autorest.typescript/issues/1456

### CODE REVIEW

**Review by joheredi — APPROVED** [review]

File: packages/rlc-codegen/src/generateSchemaType.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { ObjectDestructuringAssignment, Project } from "ts-morph";
```
Comment (lirenhe) [inline packages/rlc-codegen/src/generateSchemaType.ts:1]: Could you add license term for each of the files?

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

File: packages/cadl-rlc-emitter/src/index.ts [hunk 2]
```diff
@@ -1,55 +1,124 @@
...
-                ]
-            }
-        };
+  const rlcModels = await transformRLCModels(program);
+  const project = new Project()
+  await emitCLientDefinition(rlcModels, program);
+  await emitModels(rlcModels, program, project);
+}
+
+async function emitModels(rlcModels: RLCModel, program: Program, project: Project) {
```
Comment (lirenhe) [inline packages/cadl-rlc-emitter/src/index.ts:23]: Do we have any test cases for the current emitter?

Comment (qiaozha) [reply packages/cadl-rlc-emitter/src/index.ts]: We have one here https://github.com/Azure/autorest.typescript/blob/cadl-integration/packages/cadl-rlc-test/spec/main.cadl but we don't checkin the generated output file yet.

File: packages/cadl-rlc-emitter/src/index.ts [hunk 3]
```diff
@@ -1,34 +1,50 @@
...
+) {
+  const schemaOutput = generateSchemaTypes(rlcModels, project);
+  if (schemaOutput) {
+    const { inputModelFile, outputModelFile } = schemaOutput;
+    if (inputModelFile) await emitFile(inputModelFile, program);
+    if (outputModelFile) await emitFile(outputModelFile, program);
+  }
+}
+
+async function emitCLientDefinition(rlcModels: RLCModel, program: Program) {
```
Comment (lirenhe) [inline packages/cadl-rlc-emitter/src/index.ts:32]: typo, emitClientDefinition

Comment (qiaozha) [reply packages/cadl-rlc-emitter/src/index.ts]: updated

File: packages/rlc-codegen/src/generateSchemaType.ts (hunk 1 above)
Comment (qiaozha) [reply packages/rlc-codegen/src/generateSchemaType.ts:4]: updated.

**Pull request merged**